src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -0,0 +1,103 @@
+import scala.util.Try
+
+object CromwellApiServiceActor {
+  def props(workflowManagerActorProps : Props, swaggerService: SwaggerService): Props = {
+    Props(new CromwellApiServiceActor(workflowManagerActorProps, swaggerService))
+  }
+}
+
+class SwaggerService(override val apiVersion: String,
+                     override val baseUrl: String,
+                     override val docsPath: String,
+                     override val swaggerVersion: String,
+                     override val apiTypes: Seq[Type],
+                     override val apiInfo: Option[ApiInfo])
+                    (implicit val actorRefFactory: ActorRefFactory)
+  extends SwaggerHttpService
+
+class CromwellApiServiceActor(val workflowManagerActorProps : Props, swaggerService: SwaggerService) extends Actor with RootCromwellApiService with CromwellApiService {
+  implicit def executionContext = actorRefFactory.dispatcher
+  def actorRefFactory = context
+
+  def possibleRoutes = baseRoute ~ workflowRoutes ~ swaggerService.routes ~
+    get {
+      pathSingleSlash {
+        getFromResource("swagger/index.html")
+      } ~ getFromResourceDirectory("swagger/") ~ getFromResourceDirectory("META-INF/resources/webjars/swagger-ui/2.0.24/")
+    }
+
+  def receive = runRoute(possibleRoutes)
+
+}
+
+@Api(value = "", description = "Cromwell Base API", position = 1)
+trait RootCromwellApiService extends HttpService {
+  @ApiOperation(value = "Check if Cromwell is alive",
+    nickname = "poke",
+    httpMethod = "GET",
+    produces = "text/html",
+    response = classOf[String])
+  @ApiResponses(Array(
+    new ApiResponse(code = 200, message = "Successful Request"),
+    new ApiResponse(code = 500, message = "Cromwell Internal Error")
+  ))
+  def baseRoute = {
+    path("headers") {
+      get {
+        requestContext => requestContext.complete(requestContext.request.headers.mkString(",\n"))
+      }
+    }
+  }
+}
+
+@Api(value = "workflows", description = "APIs for Workflow Operations", position = 1)
+trait CromwellApiService extends HttpService with PerRequestCreator  {
+  val workflowManagerActorProps : Props
+
+  val workflowRoutes = queryRoute
+  // ~ listWorkspacesRoute
+
+  @ApiOperation(value = "Query for workflow based on workflow id.",
+    nickname = "query",
+    httpMethod = "GET",
+    produces = "application/json"
+  )
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name = "id", required = true, dataType = "string", paramType = "path", value = "workflow")
+  ))
+  @ApiResponses(Array(
+    new ApiResponse(code = 200, message = "Successful Request"),
+    new ApiResponse(code = 404, message = "Workflow ID Not Found"),
+    new ApiResponse(code = 500, message = "Internal Error")
+  ))
+  def queryRoute =
+    path("workflows" / Segment) { id =>
+
+      // UUID.fromString can throw an IllegalArgumentException if it's malformed
+      val workflowId: Option[WorkflowId] = Try(UUID.fromString(id)).toOption
+
+      if (workflowId == None) {
```

Comment:
Similar comment as above. If you do simply want to check for None-ness, the `isEmpty` is a little better. But since this is effectively an if/else and not just an if, the same advice as above holds